bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/CarouselImpl.java

```diff
@@ -74,4 +73,13 @@ public boolean getAutopauseDisabled() {
     public String getAccessibilityLabel() {
         return accessibilityLabel;
     }
+
+    /*
+     * DataLayerProvider implementation of field getters
+     */
+
+    @Override
+    public String[] getDataLayerShownItems() {
+        return  getItems() != null ? new String[]{getItems().get(0).getId()} : new String[0];
```

Comment:
shouldn't it be instead?
`getItems().get(0).getDataLayer().getId() `

Reply:
Done in 2509408